lib/glific/providers/gupshup/api_client.ex

```diff
@@ -2,12 +2,139 @@ defmodule Glific.Providers.Gupshup.ApiClient do
   @moduledoc """
   Http API client to intract with Gupshup
   """
+  alias Glific.Partners
   alias Plug.Conn.Query
 
+  @gupshup_url "https://api.gupshup.io/sm/api/v1"
+
   use Tesla
   # you can add , log_level: :debug to the below if you want debugging info
   plug Tesla.Middleware.Logger
 
   plug Tesla.Middleware.FormUrlencoded,
     encode: &Query.encode/1
+
+  @doc """
+  Making Tesla get call and adding api key in header
+  """
+  @spec gupshup_get(String.t(), String.t()) :: Tesla.Env.result()
+  def gupshup_get(url, api_key), do: get(url, headers: [{"apikey", api_key}])
+
+  @doc """
+  Making Tesla post call and adding api key in header
+  """
+  @spec gupshup_post(String.t(), any(), String.t()) :: Tesla.Env.result()
+  def gupshup_post(url, payload, api_key), do: post(url, payload, headers: [{"apikey", api_key}])
+
+  @spec get_credentials(non_neg_integer()) :: {:error, String.t()} | {:ok, map()}
+  defp get_credentials(org_id) do
+    organization = Partners.organization(org_id)
+
+    if is_nil(organization.services["bsp"]) do
+      {:error, "No active BSP available"}
+    else
+      bsp_credentials = organization.services["bsp"]
```

Comment:
We should check that we have values for everything, for the api_key and appname also